freqtrade/optimize/backtesting.py

```diff
@@ -467,32 +480,38 @@ def _get_sell_trade_entry_for_candle(self, trade: LocalTrade,
             ):
                 trade.sell_reason = sell_row[EXIT_TAG_IDX]
 
-            self.order_id_counter += 1
-            order = Order(
-                id=self.order_id_counter,
-                ft_trade_id=trade.id,
-                order_date=sell_candle_time,
-                order_update_date=sell_candle_time,
-                ft_is_open=True,
-                ft_pair=trade.pair,
-                order_id=str(self.order_id_counter),
-                symbol=trade.pair,
-                ft_order_side="sell",
-                side="sell",
-                order_type=order_type,
-                status="open",
-                price=closerate,
-                average=closerate,
-                amount=trade.amount,
-                filled=0,
-                remaining=trade.amount,
-                cost=trade.amount * closerate,
-            )
-            trade.orders.append(order)
-            return trade
+            return self._exit_trade(trade, sell_row, closerate)
 
         return None
 
+    def _exit_trade(self, trade: LocalTrade, sell_row: Tuple,
+                    closerate: float, amount: float = None) -> Optional[LocalTrade]:
+        self.order_id_counter += 1
+        sell_candle_time = sell_row[DATE_IDX].to_pydatetime()
+        order_type = self.strategy.order_types['sell']
+        order = Order(
+            id=self.order_id_counter,
+            ft_trade_id=trade.id,
+            order_date=sell_candle_time,
+            order_update_date=sell_candle_time,
+            ft_is_open=True,
+            ft_pair=trade.pair,
+            order_id=str(self.order_id_counter),
+            symbol=trade.pair,
+            ft_order_side="sell",
+            side="sell",
+            order_type=order_type,
+            status="open",
+            price=closerate,
+            average=closerate,
+            amount=amount or trade.amount,
+            filled=0,
+            remaining=trade.amount,
+            cost=trade.amount * closerate,
```

Comment:
if you use `amount or trade.amount` above - the same should be used for the calculation of cost.

Reply:
updated